tryy/base_class.py

```diff
@@ -0,0 +1,115 @@
+# -!- coding: utf-8 -!-
+from datetime import datetime
+from datetime import timedelta
+from datetime import timezone
+import hashlib
+
+from bs4 import BeautifulSoup
+import requests
+
+# class get_mutiple_link:
+
+#     def ___init__(self, link)
+#         self.link = link
+
+#     def get_page
+
+
+class Base:
+
+    def __init__(self,
+                 title,
+                 content,
+                 category,
+                 modified_date,
+                 media,
+                 url=None,
+                 url_hash=None,
+                 content_hash=None,
+                 headers=None):
+        self.title = title
+        self.content = content
+        self.category = category
+        self.modified_date = self.get_modified_date(modified_date)
+        self.media = media
+        # self.tags = tags
+        self.url = self.get_page(url, headers)
+        self.url_hash = url_hash if url_hash else self.generate_hash(
+            url) if url else None
+        self.content_hash = content_hash if content_hash else self.generate_hash(
+            content) if content else None
+
+    def get_page(self, url, headers):
+        try:
+            r = requests.get(url, headers)
+            r.encoding = 'UTF-8'
+            soup = BeautifulSoup(r.text, "html.parser")
+            return soup
+        except requests.RequestException as e:
+            print(f"Error fetching page")
```

Comment:
You can print out the error, and don't use f-strings if the string is hard-coded.